src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden && animal.health > 0) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        Animal.alive = Animal.alive.filter((a) => a.health > 0);
+      }
+    }
```

Comment:
[CODE PERFORMANCE] While the implementation of the `bite` method is correct, there's an opportunity to optimize the code by avoiding unnecessary checks. Since you're already filtering the `Animal.alive` array to remove dead animals, you don't need to check if `animal.health > 0` before applying damage. The filter will take care of removing any animals that have died as a result of the bite.